src/components/modal/modal.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  bind,
+  forwardRef,
+  Component,
+  ComponentRef,
+  Directive,
+  DynamicComponentLoader,
+  ElementRef,
+  Host,
+  Injectable,
+  ResolvedBinding,
+  SkipSelf,
+  Injector,
+  View,
+  ViewEncapsulation
+} from 'angular2/core';
+
+import {ObservableWrapper, Promise, PromiseWrapper} from 'angular2/src/core/facade/async';
+import {isPresent, Type} from 'angular2/src/core/facade/lang';
+import {DOM} from 'angular2/src/core/dom/dom_adapter';
+import {MouseEvent, KeyboardEvent} from 'angular2/src/core/facade/browser';
+import {KeyCodes} from 'angular2_material/src/core/key_codes';
```

Comment:
All this lines, perhaps we need to change that to `angular2/angular2` or similar. Seems that the import "source" changes depending on what angular2 package the end user have in their project or something like that. Need to research on that.